### ISSUE TITLE
VPAT 106: Missing `aria-labelledby` connector between modal container and it's title

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#18811]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

Missing `aria-labelledby` connector between modal container and it's title

```
Actual Result :
Modal dialog is not specified with a programmatic accessible name which resulted in difficulty for screen reader users to understand the purpose of a dialog.

Instances:
Change Password 
- Change Password modal dialog

Delete API Key
- Are you sure? modal dialog 

Import YAML
Import YAML modal dialog

Cluster Appearance
- Cluster Appearance modal dialog

Move
- Move to a new project modal dialog 

Delete Project
-  Are you sure? modal dialog

Add Project Member
- Add Project Member modal dialog

Add Extensions repositories
- Add Extensions repositories modal dialog

Import Extension Catalog
- Import Extension Catalog
 
Existing Code:
<div data-testid="change-password__modal" height="465" id="modal-container-element" class="change-password-modal modal-container" role="dialog" aria-modal="true" style="width: 600px;">
(...)
<h4 data-v-5df885cb="" class="text-default-text">Change Password</h4>

Recommendation:
Apply the following changes:
• Add "aria-labelledby" attribute to the <div> element containing the modal dialog content.
• Reference the "id" of the heading element containing the "Change Password" text via "aria-labelledby" attribute.

For information on creating accessible modal dialog refer https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html
 
Suggested Code:
<div data-testid="change-password__modal" height="465" id="modal-container-element" class="change-password-modal modal-container"  role="dialog" aria-modal="true" aria-labelledby="modalheading1" style="width: 600px; >
(…)
<h4 data-v-5df885cb="" id="modalheading1" class="text-default-text">Change Password</h4>
(…)
</div> 
```

**To Reproduce**

Check DOM 

**Result**
Missing `aria-labelledby` connector between modal container and it's title

**Expected Result**

Modal container and it's title should be connected by `aria-labelledby`

**Screenshots**

**Additional context**
